articles/how-to-build-astatic-site-with-gatsbyjs/index.md

```diff
@@ -0,0 +1,272 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/how-to-build-astatic-site-with-gatsbyjs/
+title: How to build a static site with GatsbyJs
+description:
+author: jethro-magaji
+date: 2020-10-19T00:00:00-08:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/how-to-build-astatic-site-with-gatsbyjs/hero.png
+    alt: hero example image computer Node.js
+---
+When I first heard about GatsbyJs, I was confused about what it was, and how I would get started. As time went on, I learned more about it and found out it is awesome. My goal for this article is not to confuse you about GatsbyJs, but to help you get started with building your very first static site using GatsbyJs.
+<!--more-->
+A paradigm shift has begun in the web development world, where static sites are now becoming more of a trend over dynamic sites. Building a dynamic site that required you to set it up and/or maintaining it, is now obsolete.
+
+This is not an issue when building static sites. Due to cloud service companies like [Amazon](https://aws.amazon.com/), [Google](https://cloud.google.com/), [Netlify](https://www.netlify.com/), [Heruko](https://www.heroku.com/), and [Gaysby Cloud](https://www.gatsbyjs.com/cloud/) etc. Having worked on making software development easier, faster, and more scalable for developers.
+
+WHen compared to static sites, dynamic sites are slow to load and can cause high bounce rate on a page by users, now that’s where GatsbyJs comes into play; it is a powerful and easy static site generator.
+
+### What is GatsbyJs?
+>GatsbyJs can be defined as a static site generator that uses [React.js](https://reactjs.org/) (for the client-side) and [GraphQL](https://graphql.org/) (to access data) to build a reliable and faster website.
+
+**Advantages**
+- It's [faster](https://www.gatsbyjs.com/contributing/how-to-pitch-gatsby/) to create a static site with GatsbyJs.
+- It's easy to set up.
+- It's scalable.
+- It has a great [community of developers](https://www.gatsbyjs.com/contributing/community/).
+
+### Prerequisites
+To get started with GatsbyJs, you will need some background knowledge on the following:
+- [React.js](https://reactjs.org/) (the basics, components, etc.)
+- [Nodejs](https://nodejs.org/en/) (the basics, [npm](https://www.npmjs.com/), etc)
+- [GraphQL](https://graphql.org/) (the basics and advanced)
+
+However there’s no problem if you don’t have any knowledge on these technologies, this article will help you get started with GatsbyJs, so stick around and check the links out later if you are interested in advancing your knowledge.
+
+### Installations
+First of all, you'll need to install [Node.js](https://nodejs.org/en/) on your computer.
+Go to the [official Node.js website](https://nodejs.org/en/) to download the Node.js version for your operating system.
+
+After installing it, open up your terminal and type in:
+
+```bash
+Users/yourname: node –version
```

Comment:
Remove Users/yourname: for all code snippets. This is not necessary.

Reply:
Resolved